### PR TITLE
DBCON-178: fixed performance regression since 3.9.4

### PR DESCRIPTION
**Pull request by nicomz**

### CODE REVIEW

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java [hunk 1]
```diff
@@ -51,6 +52,13 @@ public OracleDbConnection(Connection delegate, TransactionalAction transactional
         super(delegate, transactionalAction, connectionReleaseListener, paramTypeResolverFactory);
     }
 
+    public OracleDbConnection(Connection delegate, TransactionalAction transactionalAction, DefaultDbConnectionReleaser connectionReleaseListener, ParamTypeResolverFactory paramTypeResolverFactory,Map<String, Map<Integer, ResolvedDbType>>
+            resolvedDbTypesCache)
+    {
+        super(delegate, transactionalAction, connectionReleaseListener, paramTypeResolverFactory);
+        this.resolvedDbTypesCache =resolvedDbTypesCache;
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java:59]: entire class is badly formatted.

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java [hunk 2]
```diff
@@ -42,6 +42,7 @@
 
     private Method createArrayMethod;
     private boolean initialized;
+    private Map<String, Map<Integer, ResolvedDbType>> resolvedDbTypesCache= new HashMap<>();
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java:45]: this cache should never be created here. It must always be received in the constructor

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java [hunk 3]
```diff
@@ -131,6 +140,20 @@ protected void resolveLobs(String typeName, Object[] attributes, TypeResolver ty
...
+        if(this.resolvedDbTypesCache.containsKey(typeName)){
+            if(logger.isDebugEnabled()) {
+                logger.info("Returning chached LobFieldsDataTypeInfo");
+            }
+            return resolvedDbTypesCache.get(typeName);
+        }
+        if(logger.isDebugEnabled()){
+            logger.info("Obtaining LobFieldsDataTypeInfo");
+        }
+        synchronized (this){
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java:153]: the map should be the monitor

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleTransactionalDbConnectionFactory.java [hunk 4]
```diff
@@ -22,6 +25,7 @@
  */
 public class OracleTransactionalDbConnectionFactory extends TransactionalDbConnectionFactory
 {
+    Map<String, Map<Integer, ResolvedDbType>> resolvedDbTypesCache= new HashMap<>();
```
Comment (marianogonzalez) [inline modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleTransactionalDbConnectionFactory.java:28]: ConcurrentHashMap

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java (hunk 2 above)
Comment (nicomz) [reply modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java]: Done

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java (hunk 3 above)
Comment (nicomz) [reply modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java]: Done

File: modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleTransactionalDbConnectionFactory.java (hunk 4 above)
Comment (nicomz) [reply modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleTransactionalDbConnectionFactory.java]: Done

**Pull request merged**